### PR TITLE
Feature/9638 start from store summary step

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: #9638 

### Description

Updates store creation navigation to start from Store Summary step.
In the next PR (to be merged after this one) the rest of the steps reordering is done. 

### Testing instructions

1. Trigger store creation flow and verify that store summary step is the first one.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt [hunk 1]
```diff
@@ -45,7 +48,6 @@ class StoreCreationSummaryViewModel @Inject constructor(
                 AnalyticsTracker.KEY_STEP to AnalyticsTracker.VALUE_STEP_STORE_SUMMARY
             )
         )
-
         val newStoreProfilerData = newStore.data.profilerData
         tracker.track(
             stat = AnalyticsEvent.SITE_CREATION_PROFILER_DATA,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: Should we keep this event in the new flow? if yes, then we need to move it somewhere else, as at this step, we won't have any profiler data, we can probably move it to the `StoreProfilerRepository#uploadAnswers` to be called at the same time where we will send the wc-admin request, WDYT? If we agree, I can even add it to my PR #9633

**Review by hichamboushaba — APPROVED** [review]
Looks good, thanks @JorgeMucientes 
I just left one question to discuss how we should approach the profiler tracks event, let's discuss it before merging this.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: Great catch @hichamboushaba!! I see 2 options here: 
- We can track the profiler data inside `storeProfilerRepository.uploadAnswers()` so everything is handled together. 
- Or we can track the profiler data from `StoreInstallationViewModel` once we know all the profiler steps are done. 

I lean towards the first approach where everything is kept in the same place. Wdyt? 

For now, I'll remove this tracking code from this class in this PR.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: Same, I lean towards the first approach, as with this we can still be able to upload the partial data we got even if the user didn't finish the steps, as the site will still be created in the background.

> For now, I'll remove this tracking code from this class in this PR.

You can keep it, I'll move it in my PR, just to have a clear history of what happened to the code.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: @JorgeMucientes can you revert the last commit for the reason I explained above? 🙂
I implemented the change fully in my PR https://github.com/woocommerce/woocommerce-android/commit/999fbb0cf457c3869600731431e4de684e7d3f93

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: ~~Ouch! Sorry @hichamboushaba I didn't see the reply in time. And I committed the tracking removal. I can reset the commit if you prefer to keep a clean history, but I think is a minor thing.~~

I've git reseted the change.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/summary/StoreCreationSummaryViewModel.kt]: Personally, I think that even though they are minor points, having a clear history helps in understanding the reasons behind changes, but it's not a blocking point, you can keep it 🙂.

edit: thanks for removing it 🥇

**Pull request merged**